spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/config/SkipperServerConfiguration.java

```diff
@@ -172,9 +188,51 @@ public AppDeployerReleaseManager appDeployerReleaseManager(ReleaseRepository rel
 			AppDeployerDataRepository appDeployerDataRepository,
 			DeployerRepository deployerRepository,
 			ReleaseAnalyzer releaseAnalyzer,
-			AppDeploymentRequestFactory appDeploymentRequestFactory) {
+			AppDeploymentRequestFactory appDeploymentRequestFactory,
+			UpgradeStrategy updateStrategy) {
 		return new AppDeployerReleaseManager(releaseRepository, appDeployerDataRepository, deployerRepository,
-				releaseAnalyzer, appDeploymentRequestFactory);
+				releaseAnalyzer, appDeploymentRequestFactory, updateStrategy);
+	}
+
+	@Bean
+	public DeleteStep deleteStep(ReleaseRepository releaseRepository,
+			DeployerRepository deployerRepository,
+			AppDeployerDataRepository appDeployerDataRepository) {
+		return new DeleteStep(releaseRepository, deployerRepository, appDeployerDataRepository);
+	}
+
+	@Bean
+	public UpgradeStrategy updateStrategy(ReleaseRepository releaseRepository,
+			DeployerRepository deployerRepository,
+			AppDeployerDataRepository appDeployerDataRepository,
+			AppDeploymentRequestFactory appDeploymentRequestFactory,
+			HealthCheckAndDeleteStep healthCheckAndDeleteStep) {
+		return new SimpleRedBlackUpgradeStrategy(releaseRepository, deployerRepository,
+				appDeployerDataRepository, appDeploymentRequestFactory, healthCheckAndDeleteStep);
+	}
+
+	@Bean
+	public HealthCheckAndDeleteStep healthCheckAndDeleteStep(ReleaseRepository releaseRepository,
+			DeployerRepository deployerRepository,
+			AppDeployerDataRepository appDeployerDataRepository,
+			DeleteStep deleteStep,
+			HealthCheckProperties healthCheckProperties) {
+		return new HealthCheckAndDeleteStep(releaseRepository,
+				deployerRepository,
+				deleteStep,
+				healthCheckProperties);
+	}
+
+	@Bean(name = "skipperThreadPoolTaskExecutor")
+	@Override
+	public Executor getAsyncExecutor() {
+		return new ThreadPoolTaskExecutor();
+	}
+
+	@Override
+	public AsyncUncaughtExceptionHandler getAsyncUncaughtExceptionHandler() {
```

Comment:
Right ok, this might explain where execution errors end up what I mentioned in other comments.

Reply:
In my testing, this never got called.  That is why I put a large catch block around all of the code in the `@Async` method.  Not sure what I am doing wrong.